File: easy/cielandreceipt/cielandreceipt.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

# started at: Saturday, 14. December 2013 09:14PM
import sys

def find_j(num,menus):
    for i in menus:
        if num<i:
            return menus[menus.index(i)-1]
    return menus[11]

def main():
    iterations = int(sys.stdin.readline())
    menus = [1,2,4,8,16,32,64,128,256,512,1024,2048]

    for i in xrange(iterations):
        num = int(sys.stdin.readline())

        if num in menus:
            print '1'
            continue

        count = 0
        while num!=0:
            j = find_j(num,menus)
            num -= j
            count += 1
        print count

if __name__ == '__main__':
    main()

File: easy/cielandreceipt/cielandreceipt.c
/*
    started at: Saturday, 14. December 2013 09:14PM
*/

#include <stdio.h>


int find_k(int num,int array[])
{
    for (int i=0; i<12; ++i)
    {
        if (array[i]>num)
        {
            return array[i-1];
        }
    }
    return array[11];
}
int main(int argc, char const *argv[])
{
    int iterations = 0;
    scanf("%d",&iterations);

    int menus[] = {1,2,4,8,16,32,64,128,256,512,1024,2048};

    int count = 0;
    int num = 0;
    int k = 0;

    for (int i=0; i<iterations; ++i)
    {
        count = 0;
        scanf("%d",&num);

        for (int j=0; j<12; ++j) 
        {
            if (menus[j]==num)
            {
                count = 1;        
            }
        }
        if (count!=0)
        {
            printf("%d\n",count);
            continue;
        }
        while (num!=0) 
        {
            k = find_k(num,menus);
            num -= k;
            count++;             
        }
        printf("%d\n",count);
    }

    return 0;
}